aws/resource_aws_ssm_resource_data_sync.go

```diff
@@ -0,0 +1,160 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsSsmResourceDataSync() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSsmResourceDataSyncCreate,
+		Read:   resourceAwsSsmResourceDataSyncRead,
+		Delete: resourceAwsSsmResourceDataSyncDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"s3_destination": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"kms_key_arn": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"bucket_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"prefix": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"region": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"sync_format": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  ssm.ResourceDataSyncS3FormatJsonSerDe,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsSsmResourceDataSyncCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ssmconn
+
+	input := &ssm.CreateResourceDataSyncInput{
+		S3Destination: expandSsmResourceDataSyncS3Destination(d),
+		SyncName:      aws.String(d.Get("name").(string)),
+	}
+
+	_, err := conn.CreateResourceDataSync(input)
+	if err != nil {
+		return err
+	}
+	d.SetId(d.Get("name").(string))
+	return resourceAwsSsmResourceDataSyncRead(d, meta)
+}
+
+func resourceAwsSsmResourceDataSyncRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ssmconn
+
+	syncItem, err := findResourceDataSyncItem(conn, d.Get("name").(string))
+	if err != nil {
+		if _, ok := err.(awserr.Error); ok {
+			return err
+		} else {
+			d.SetId("")
+			return nil
+		}
```

Comment:
I'm slightly worried about this conditional here, because we may add more errors to `findResourceDataSyncItem` which aren't `awserr` and those may have nothing to do with "resource not found".
How do you feel about tweaking `findResourceDataSyncItem` to return `nil, nil` if no resource was found and then checking the `nil` here, instead of checking the error _isn't_ awserr.Error?

Reply:
Sure👍 I'll apply that idea!